Coop Spreadsheet View - Resolves #190

domdelorenzo opened the pull request:
Updated spreadsheet component (available at http://localhost:3001/spreadsheet in docker) now spawns an update modal window that can update Coop records. 

Outstanding question:
- How do we want to display data for the "proposed changes" column? (it's currently a json object - do we want to stringify it in the spreadsheet?)

laredotornado commented:
Hi, Answer is, we don't.  "proposed changes" is only for people who submit
coop edit/add data for an admin to approve.  THis column should definitely
be suppressed in the spreadsheet view.

Thanks! -

On Wed, Jun 15, 2022 at 7:11 PM domdelorenzo ***@***.***>
wrote:

> Updated spreadsheet component (available at
> http://localhost:3001/spreadsheet in docker) now spawns an update modal
> window that can update Coop records.
>
> Outstanding question:
>
>    - How do we want to display data for the "proposed changes" column?
>    (it's currently a json object - do we want to stringify it in the
>    spreadsheet?)
>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/chicommons/maps/pull/200
> Commit Summary
>
>    - 4c76989
>    <https://github.com/chicommons/maps/pull/200/commits/4c769898dd14154e3486233a5918b5e9d2c64cdf>
>    Added initial reactgrid test setup
>    - 8cfb1d9
>    <https://github.com/chicommons/maps/pull/200/commits/8cfb1d9df5c26bd473153d6d0b8644d915f19867>
>    Added basic grid with filter with AG-grid
>    - 503c5c9
>    <https://github.com/chicommons/maps/pull/200/commits/503c5c987a4c2fccb710dfc996a22694812bfa7d>
>    Working AG Grid component with basic filter
>    - a7de216
>    <https://github.com/chicommons/maps/pull/200/commits/a7de216a733f343fb7d0765b0c0390472248034b>
>    Merge branch 'authentication' into spreadsheet-view
>    - cced894
>    <https://github.com/chicommons/maps/pull/200/commits/cced8942178965775822a106aed8705c390106d2>
>    Renamed AGGrid component. Added Handsontable component
>    - 959c541
>    <https://github.com/chicommons/maps/pull/200/commits/959c541c1874ec3d94376b7fedd7043b276929f5>
>    Added modal popup for grid editing. Added view to fetch additional fields.
>    Added approved field to CoopSerializer
>    - 087107a
>    <https://github.com/chicommons/maps/pull/200/commits/087107affbbe518c602c777ed108ed9ab948d3b5>
>    Added Ka Table with filtering and sorting.
>    - 7e17b3c
>    <https://github.com/chicommons/maps/pull/200/commits/7e17b3cdb41dd9e64fbafcafd558ead81007abc7>
>    Added row select event to Ka-Table
>    - fcf9c48
>    <https://github.com/chicommons/maps/pull/200/commits/fcf9c4880e15b8800250305185cbd8ddfbc5bb05>
>    Added modal poppup to ka-table
>    - 1d88335
>    <https://github.com/chicommons/maps/pull/200/commits/1d88335324ca16723faadc46fe5688a37127def0>
>    Added Toast Grid
>    - 3300571
>    <https://github.com/chicommons/maps/pull/200/commits/3300571bfa32178ca654555a18744e8e94d88f50>
>    Modal window triggered by doubleclick on Toast Grid
>    - f5b2758
>    <https://github.com/chicommons/maps/pull/200/commits/f5b275887103f9f0eea9201ec2a240f7d38ae8e8>
>    Modal window can now submit updates to database
>    - b4ef3c8
>    <https://github.com/chicommons/maps/pull/200/commits/b4ef3c825dc44a3f9d9898b06ee841d8208503e0>
>    Removed unused spreadsheet componenets. Reneamed toastgrid to spreadsheet
>    and updated routes.
>
> File Changes
>
> (11 files <https://github.com/chicommons/maps/pull/200/files>)
>
>    - *M* apache/my-vhosts.conf
>    <https://github.com/chicommons/maps/pull/200/files#diff-48834454aa5659d909f36fa3f2fa575d1dd2bb2865824c488aea9acabd2b88b4>
>    (2)
>    - *M* client/package.json
>    <https://github.com/chicommons/maps/pull/200/files#diff-1846122c2c83a486a3693f7966aa522c34cf489f674185c4da0d9221683fd81f>
>    (8)
>    - *M* client/src/App.css
>    <https://github.com/chicommons/maps/pull/200/files#diff-32d1dced60f884eceee5905cccdf17256fe157a222e66e7863e8c5c6e9ea9cd4>
>    (6)
>    - *M* client/src/components/DirectoryApp.jsx
>    <https://github.com/chicommons/maps/pull/200/files#diff-650199f110a4f990166537fe3509fed761a038b0721e04f4bedacade08cf3cfe>
>    (2)
>    - *A* client/src/components/ModalUpdate.jsx
>    <https://github.com/chicommons/maps/pull/200/files#diff-aa4e2b7899b790aea0afaf3f02b8f13c0ebdd6f2d3d3d3a9fea9f2063508a6ed>
>    (765)
>    - *M* client/src/components/Search.jsx
>    <https://github.com/chicommons/maps/pull/200/files#diff-8a07ca69dc41f3d796d47cfbd1aaa90c96ac833ac8d4484e15bec5466274a80b>
>    (2)
>    - *A* client/src/components/Spreadsheet.jsx
>    <https://github.com/chicommons/maps/pull/200/files#diff-43e13c83d2434ddb9a5b9374a4d714d607d6d120893ff0ca16b7445cd59a95a9>
>    (109)
>    - *M* client/tsconfig.json
>    <https://github.com/chicommons/maps/pull/200/files#diff-ddbaeb6c44d45c91a5ffc990e6f821910381211d1b4a52d5cf58ee8023e74745>
>    (3)
>    - *M* web/directory/serializers.py
>    <https://github.com/chicommons/maps/pull/200/files#diff-f93b1e5638378c106a2e38899b9c9aacdf5f49252eea7d3e91fbd0799463e303>
>    (2)
>    - *M* web/directory/urls.py
>    <https://github.com/chicommons/maps/pull/200/files#diff-b084f36a9074782bcda661540afddd5f7ce3b2b2a12424205bd44f4e090aa20b>
>    (1)
>    - *M* web/directory/views.py
>    <https://github.com/chicommons/maps/pull/200/files#diff-cb96d715ce61cb278bcdbc3698deade309d80d10a885f0b56074a85e625eda60>
>    (40)
>
> Patch Links:
>
>    - https://github.com/chicommons/maps/pull/200.patch
>    - https://github.com/chicommons/maps/pull/200.diff
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/chicommons/maps/pull/200>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADRHTSLQIFQXFATRPDVRGADVPJWK7ANCNFSM5Y5BBHLQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

laredotornado commented on apache/my-vhosts.conf:
```diff
@@ -19,6 +19,8 @@
     ProxyPass /login http://web:8000/login
     ProxyPassReverse /login http://web:8000/login
+    ProxyPass /login http://web:8000/all
+    ProxyPassReverse /login http://web:8000/all
```
For these two lines, you'll want to change "/login" to "/all", making this

```
    ProxyPass /all http://web:8000/all
    ProxyPassReverse /all http://web:8000/all
```

domdelorenzo replied:
Updated, although I'm not sure how to test this one.. How do you access the proxy when it's served by docker?

laredotornado commented on client/src/components/Spreadsheet.jsx:
```diff
@@ -0,0 +1,109 @@
+  const [selectedCoop, setSelectedCoop] = useState([])
+  const handleClose = () => {
+    setShow(false);
+    setSelectedCoop([])
```
Minor thing, but I notice sometimes you use semicolons at the end of lines and other times not, any specific reason?

domdelorenzo replied:
Thanks for calling that out - that's sloppy on my part. I typically use a vscode extension to auto-add semicolons at the end of lines, but it doesn't seem to be working.

laredotornado reviewed (commented):
Great PR!  Since we already talked about excluding the "proposed_changes" from teh serializer, I'll wait to approve until you have that in.

domdelorenzo commented:
@laredotornado, I removed "proposed_changes" from the serializer and fixed some of the sloppy formatting. I've also updated the apache config, but I'm not sure how to test it. Let me know what I should be looking at and I'll take a look!

laredotornado commented:
Sadly right now, after updating the Apache configs you have to rebuild the
docker containers

```
docker-compose down --rmi all
docker-compose up
```

After that's done, you should be able to interact with your spreadsheet

```
http://loaclhost:3001/spreadsheet
```

or hit the new endpoint directly

```
http://localhost:9090/all
```

I can also give it a whirl

On Thu, Jun 16, 2022 at 8:40 PM domdelorenzo ***@***.***>
wrote:

> @laredotornado <https://github.com/laredotornado>, I removed
> "proposed_changes" from the serializer and fixed some of the sloppy
> formatting. I've also updated the apache config, but I'm not sure how to
> test it. Let me know what I should be looking at and I'll take a look!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/chicommons/maps/pull/200#issuecomment-1158381893>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADRHTSJSY2ERXZLRBW6IAPTVPPJR5ANCNFSM5Y5BBHLQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
